interfaces/builtin/desktop_launch.go

```diff
@@ -90,6 +90,7 @@ func (iface *desktopLaunchInterface) Name() string {
 func (iface *desktopLaunchInterface) StaticInfo() interfaces.StaticInfo {
 	return interfaces.StaticInfo{
 		Summary:              desktopLaunchSummary,
+		ImplicitOnCore:       true,
```

Comment:
I believe this would only make sense for core desktop? even though it is not yet supported which is my other concern.
```suggestion
		ImplicitOnCore:       release.OnCoreDesktop,
```

Reply:
The thing is, most SnapD interfaces only allow access to resources, whether they exist on a system or not. As documented above, you **can** have a user session, with userd running, on Core. You just don't, by default. Same if you install Ubuntu Server.

IMO the reason to split interfaces out is only if you want to allow a subset of what the existing one provides - not whether they're used or not. So yes, if you wanted a snap to only have access to app metadata, but not allowing it to launch them, then you'll need separate interfaces. Otherwise we'll grow the number of interfaces quicker than necessary.